chain/network/src/peer_manager/peer_store.rs

```diff
@@ -85,23 +85,25 @@ impl PeerStore {
             let mut peer_state: KnownPeerState = value.try_into()?;
             // Mark loaded node last seen to now, to avoid deleting them as soon as they are loaded.
             peer_state.last_seen = to_timestamp(Utc::now());
-            match peer_state.status {
-                KnownPeerStatus::Banned(_, _) => {}
-                _ => peer_state.status = KnownPeerStatus::NotConnected,
+            peer_state.status = match peer_state.status {
+                banned_status @ KnownPeerStatus::Banned(_, _) => banned_status,
+                _ => KnownPeerStatus::NotConnected,
             };
 
-            if let Some(current_peer_state) = peer_states.get_mut(&peer_id) {
-                // This peer is a boot node and was already added so skip.
-                if peer_state.status.is_banned() {
-                    current_peer_state.status = peer_state.status;
+            match peer_states.entry(peer_id) {
+                Entry::Occupied(mut current_peer_state) => {
+                    // This peer is a boot node and was already added so skip.
+                    if peer_state.status.is_banned() {
+                        current_peer_state.get_mut().status = peer_state.status;
+                    }
                 }
-                continue;
-            }
-
-            if let Some(peer_addr) = peer_state.peer_info.addr {
-                if let Entry::Vacant(entry) = addr_peers.entry(peer_addr) {
-                    entry.insert(VerifiedPeer::new(peer_state.peer_info.id.clone()));
-                    peer_states.insert(peer_id, peer_state);
+                Entry::Vacant(entry) => {
+                    if let Some(peer_addr) = peer_state.peer_info.addr {
```

Comment:
so if 2 peers are at the same address, we're going to add only one of them ? Shouldn't we somehow pick the "newer" one ? or it doesn't matter (as if we pick the old one - it will be marked as unreachable in 1h.. )

Reply:
@mm-near This doesn't matter for bootnodes, as all of them are considered new.
There should be only one existing in the database.

Reply:
There are a few cases:
- multiple peers have the same address in `boot_nodes` - we probably should panic!
- there is an entry for a peer in boot nodes, and it's different from entry in the database.
We currently, prefer the entry in boot_nodes. There is a bug, first_seen, seems to be wrongly updates.
- We assume that database entries are unique.